app/helpers/categories_helper.rb

```diff
@@ -1,2 +1,64 @@
 module CategoriesHelper
+
+
+  def get_most_popular_categories
+    popular_categories = []
+    rec_array = AhoyEvent.where(name: "Category selected").where("time > ?", Time.now-90.days).select("properties").map { |e| e.properties["category_id"] }
+    categories_count = Hash.new(0)
+    rec_array.each { |rec| categories_count[rec] +=1 }
+    pop_cats = categories_count.sort_by { |rec,number| number}.last(20).reverse
+    pop_cats.each do |pop_cat|
+      popular_categories << Category.find_by_id(pop_cat[0]).name
+    end
+    popular_categories
+  end
+
+  def update_category_usages
+    s_query = ActiveRecord::Base.sanitize_sql_like(params["query"])
+    cat_rec = Category.where("name ILIKE ?", s_query.downcase).not_other.first
+    return if cat_rec.blank?
+    cat_id = cat_rec.id
+    last_ahoy_event = AhoyEvent.where(name: 'Category selected').last(1)
+    if last_ahoy_event.blank?
+      store_chosen_categories(s_query, nil) unless s_query.blank?
+    else
+      event_tm = last_ahoy_event[0].time
+      ev_user_id = last_ahoy_event[0].user_id
+      last_ev_cat_id = last_ahoy_event[0].properties["category_id"]
+      same_cat_id = cat_id == last_ev_cat_id
+      same_visit = (DateTime.now.to_time.utc - event_tm) <= 2
+      same_user = current_user.present? ? current_user.id == ev_user_id : ev_user_id.blank?
+      if !(same_cat_id && same_visit && same_user)
+        store_chosen_categories(s_query, nil) unless s_query.blank?
+      end
+    end
+  end
+
+  def store_chosen_categories(s_query, chosen_categories)
+    s_query = s_query.downcase
+    cats = []
```

Comment:
maybe you can rename this to `cat_names` so it is more clear and you can set it equal to `Category.not_other.pluck(:name).map { |cat| cat.downcase }`

Reply:
Yes - much cleaner.